TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/CaptureProxy.java

```diff
@@ -375,4 +420,35 @@ public static void main(String[] args) throws InterruptedException, IOException
         // work in the background.
         proxy.waitForClose();
     }
+
+    static ProxyChannelInitializer buildProxyChannelInitializer(RootCaptureContext rootContext,
+                                                                BacksideConnectionPool backsideConnectionPool,
+                                                                Supplier<SSLEngine> sslEngineSupplier,
+                                                                @NonNull RequestCapturePredicate headerCapturePredicate,
+                                                                List<String> headerOverrides,
+                                                                IConnectionCaptureFactory connectionFactory)
+    {
+        var headers = convertPairListToMap(headerOverrides);
+        return new ProxyChannelInitializer(
+            rootContext,
+            backsideConnectionPool,
+            sslEngineSupplier,
+            connectionFactory,
+            headerCapturePredicate
+        ) {
+            @Override
+            protected void initChannel(@NonNull SocketChannel ch) throws IOException {
+                super.initChannel(ch);
+                for (var kvp : headers.entrySet()) {
+                    var lineBytes = (kvp.getKey() + ": " + kvp.getValue()).getBytes(StandardCharsets.UTF_8);
+                    ch.pipeline().addAfter(ProxyChannelInitializer.CAPTURE_HANDLER_NAME, "AddHeader-" + kvp.getKey(),
```

Comment:
With multiple AddHeaders, this will reverse the output of them, should we reverse iterate through the map instead?

Reply:
Only relevant if I'm using a LinkedHashMap, so yeah, I'll make the change.